network/src/connectivity_manager/mod.rs

```diff
@@ -338,6 +340,25 @@ where
             })
             .collect();
 
+        // Limit the number of dialed connections from a Full Node
+        // This does not limit the number of incoming connections
+        // It enforces that a full node cannot have more outgoing connections than `MAX_FULLNODE_CONNECTIONS`
+        // including in flight dials.
+        let to_connect_size = match self.network_context.role() {
+            RoleType::FullNode => min(
+                to_connect.len(),
+                MAX_FULLNODE_CONNECTIONS - self.connected.len() - self.dial_queue.len(),
+            ),
+            _ => to_connect.len(),
+        };
+
+        // If there's nothing to dial, let's just skip the rest of the function
+        if to_connect_size == 0 {
+            return;
+        }
+
+        let to_connect = &to_connect[0..to_connect_size];
+
         // We tune max delay depending on the number of peers to which we're not connected. This
         // ensures that if we're disconnected from a large fraction of peers, we keep the retry
         // window smaller.
```

Comment:
I wonder if this was ever evaluated... it seems highly unlikely that we need this any more... @phlip9 any thoughts if we could delete it? Especially with the recent change that you added. If we cannot connect to a peer, being more aggressive won't make it better...

Reply:
I really wanted to delete it.

Having that kind of strange variability in the timeoff backing causes:
1. operational confusion because you don't know really what's going on without a full picture
2. like you said being more aggressive won't help, especially in a lot of cases you could have some sort of retry storm which might cause things to be worse.

Reply:
Moved to here: https://github.com/libra/libra/issues/4651

Reply:
yeah, i think this PR will probably break this and effectively set the backoff super low.